common/src/main/java/com/wynntils/features/user/tooltips/ItemStatInfoFeature.java

```diff
@@ -123,15 +131,90 @@ public void onTooltipPre(ItemTooltipRenderEvent.Pre event) {
+            } else if (KeyboardUtils.isKeyDown(GLFW.GLFW_KEY_LEFT_CONTROL)) {
+                return getRerollSuffix(style, statActualValue, possibleValues);
+            } else {
+                return getPercentSuffix(style, statActualValue, possibleValues);
+            }
+        }
 
-    private MutableComponent getDefectiveName(String itemName) {
-        return ComponentUtils.makeObfuscated("Defective " + itemName, obfuscationChanceStart, obfuscationChanceEnd);
+        private MutableComponent getInnerRollSuffix(
+                GearTooltipStyle style, StatActualValue statActualValue, StatPossibleValues possibleValues) {
+            MutableComponent rangeTextComponent = Component.literal(" <")
+                    .append(Component.literal(statActualValue.internalRoll().low() + "% to "
+                                    + statActualValue.internalRoll().high() + "%")
+                            .withStyle(ChatFormatting.GREEN))
+                    .append(">")
+                    .withStyle(ChatFormatting.DARK_GREEN);
+
+            return rangeTextComponent;
+        }
+
+        private MutableComponent getRangeSuffix(
+                GearTooltipStyle style, StatActualValue actualValue, StatPossibleValues possibleValues) {
+            Pair<Integer, Integer> displayRange =
+                    StatCalculator.getDisplayRange(possibleValues, style.showBestValueLastAlways());
+
+            MutableComponent rangeTextComponent = Component.literal(" [")
+                    .append(Component.literal(displayRange.a() + ", " + displayRange.b())
+                            .withStyle(ChatFormatting.GREEN))
+                    .append("]")
+                    .withStyle(ChatFormatting.DARK_GREEN);
+
+            return rangeTextComponent;
+        }
+
+        private MutableComponent getRerollSuffix(
+                GearTooltipStyle style, StatActualValue actualValue, StatPossibleValues possibleValues) {
+            MutableComponent rerollChancesComponent = Component.literal(String.format(
+                            Locale.ROOT, " \u2605%.2f%%", StatCalculator.getPerfectChance(possibleValues)))
+                    .withStyle(ChatFormatting.AQUA)
+                    .append(Component.literal(String.format(
+                                    Locale.ROOT,
+                                    " \u21E7%.1f%%",
+                                    StatCalculator.getIncreaseChance(actualValue, possibleValues)))
+                            .withStyle(ChatFormatting.GREEN))
+                    .append(Component.literal(String.format(
+                                    Locale.ROOT,
+                                    " \u21E9%.1f%%",
+                                    StatCalculator.getDecreaseChance(actualValue, possibleValues)))
+                            .withStyle(ChatFormatting.RED));
+
+            return rerollChancesComponent;
+        }
+
+        private MutableComponent getPercentSuffix(
```

Comment:
there doesn't seem to be any handling for values that are above/below the API range? these IDs should have the NEW tag applied and prevent the item's overall percentage from being calculated

Reply:
That was the handling I kind of removed in Part 1 since you claimed that we could not lie outside the API range. ;-)

But sure, I can add something back in that replaces "illegal" values with the text "NEW".